docs/development/design/exit-node.md

```diff
@@ -0,0 +1,145 @@
+
+## Proposal
+
+The initial scope of work if for Linux only. There are two main components, the exit node which originates all traffic leaving the Nexodus mesh and the exit node client that subscribes to the exit node service.
+
+### Exit Node
+
+- An exit-node can be defined from the startup by specifying a child prefix network to advertise by passing the default route cidr notation of `0.0.0.0`.
+- One additional flag is required to make sure a user intends on originating a default route that will forward traffic from a nexd driver interface such as `wg0` to another port such as `eth0`. The `--exit-node` flag also informs the agent to perform source NAT for traffic leaving the Nexodus mesh to allow return traffic to return to the originating exit-node and then sent back to the pre-nat source device.
+- To begin with, there will be at most one active exit node per organization (or once a VPC/slice capability is implemented). Looking forward, the exit-node client should have the ability to choose from multiple exit nodes in an organization or slice of an organization and choose from multiple exit nodes based on whatever the user's criteria is such as latency or geolocation. Since these are default routes that overlap being advertised to the mesh, the driver implementation will dictate whether more than one exit node can be concurrently active. In the case of the Wireguard driver, a peer's public key is associated to the route, so it is not possible to overlap prefixes.
+- The exit node startup would look like the following example:
+
+```shell
+ nexd router --child-prefix 0.0.0.0/0 --exit-node https://try.nexodus.127.0.0.1.nip.io
+```
+
+### Exit Node Client
+
+- In order for a client to use the exit node, the user will need to explicitly subscribe to the exit node. Since exit nodes can be disruptive to the user's connection by tunnelling all traffic to the exit node, it makes sense to have the extra step needed to explicitly join an exit node. The user can join via `nexctl` or via the web UI.
+
+An example workflow of an agent node looks like the following (open to adding client to the command):
+
+```shell
+# exit node reflexive address = x.227.102.183
+# exit node client reflexive address = x.31.152.254
+
+# Before enabling exit-node on the client
+$ wget -qO - icanhazip.com
+x.31.152.254
+
+$ sudo nexctl nexd exit-node enable
+Successfully enabled exit node client on this device
+
+$ sudo nexctl nexd exit-node list
+ENDPOINT ADDRESS         PUBLIC KEY
+x.227.102.183:44728     6/CwH/gzz9jdKnxeVpWFivycFQMBkniLeBNFQq0+f04=
+
+$ wget -qO - icanhazip.com
+x.227.102.183
+
+$ sudo nexctl nexd exit-node disable
+Successfully disabled exit node client on this device
+
+# After subscribing to the exit-node, the client is now originating traffic from the exit node.
```

Comment:
minor thing but im guessing this comment should be on line 42 after we've run `sudo nexctl nexd exit-node enable`.

At this point it looks like we've unsubscribed from the exit node

Reply:
Thanks, done!